x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -547,6 +554,41 @@ private PhysicalOperation planHashJoin(HashJoinExec join, LocalExecutionPlannerC
         return source.with(new ProjectOperatorFactory(projection), layout);
     }
 
+    /**
+     * Convert a {@link HashJoinExec} into an {@link EvalOperatorFactory eval}.
+     * This is mostly useful when we're merging a table without any lookup fields
+     * and only works for single-row {@link LocalRelation}s.
+     */
+    private PhysicalOperation planHashJoinAsEval(PhysicalOperation source, HashJoinExec join, LocalExecutionPlannerContext context) {
```

Comment:
This doesn't change LOOKUP at all, but it's conceivable we could use LOOKUP on a single row table and make this go.

I did try implementing this by planning an `EvalExec` instead of a `HashJoinExec` but that was a bit more complicated. We could make it work, but this was simpler. I'm not sure what's better to be honest.

Reply:
I agree that, conceptually, this should really be a physical optimization. But if it's simpler, it's simpler :)